Microsoft.Toolkit.Uwp.UI.Controls/TextBoxMask/TextBoxMask.cs

```diff
@@ -0,0 +1,303 @@
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using Windows.ApplicationModel.DataTransfer;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// TextBox Mask property allows a user to more easily enter fixed width text in TextBox control
+    /// where you would like them to enter the data in a certain format
+    /// </summary>
+    public partial class TextBoxMask
+    {
+        private const string DefaultPlaceHolder = "_";
+        private static readonly KeyValuePair<char, string> AlphaCharacterRepresentation = new KeyValuePair<char, string>('a', "[A-Za-z]");
+        private static readonly KeyValuePair<char, string> NumericCharacterRepresentation = new KeyValuePair<char, string>('9', "[0-9]");
+        private static readonly KeyValuePair<char, string> AlphaNumericRepresentation = new KeyValuePair<char, string>('*', "[A-Za-z0-9]");
+
+        private static void OnMaskChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            InitTextBoxMask(d, e);
+        }
+
+        private static void OnPlaceHolderChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            InitTextBoxMask(d, e);
+        }
+
+        private static void OnCustomMaskChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            InitTextBoxMask(d, e);
+        }
+
+        private static void InitTextBoxMask(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var textbox = d as TextBox;
+
+            // incase no value is provided us it as normal textbox
+            var mask = textbox?.GetValue(MaskProperty) as string;
+            if (string.IsNullOrWhiteSpace(mask))
+            {
+                return;
+            }
+
+            var placeHolderValue = textbox.GetValue(PlaceHolderProperty) as string;
+            if (string.IsNullOrEmpty(placeHolderValue))
+            {
+                return;
+            }
+
+            var placeHolder = placeHolderValue[0];
+
+            var representationDictionary = textbox.GetValue(RepresentationDictionaryProperty) as Dictionary<char, string>;
+            representationDictionary = new Dictionary<char, string>();
```

Comment:
Shouldn't there be a check for null on the `representationDictionary ` before creating it.